src/coll/options.rs

```diff
@@ -29,58 +43,126 @@ pub enum WriteModel {
     ReplaceOne {
         filter: bson::Document,
         replacement: bson::Document,
-        upsert: bool,
+        upsert: Option<bool>,
     },
     UpdateOne {
         filter: bson::Document,
         update: bson::Document,
-        upsert: bool,
+        upsert: Option<bool>,
     },
     UpdateMany {
         filter: bson::Document,
         update: bson::Document,
-        upsert: bool,
+        upsert: Option<bool>,
     },
 }
 
 /// Options for aggregation queries.
-#[derive(Clone, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct AggregateOptions {
-    pub allow_disk_use: bool,
-    pub use_cursor: bool,
+    pub allow_disk_use: Option<bool>,
+    pub use_cursor: Option<bool>,
     pub batch_size: i32,
     pub max_time_ms: Option<i64>,
     pub read_preference: Option<ReadPreference>,
 }
 
+impl AggregateOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl From<AggregateOptions> for bson::Document {
+    fn from(options: AggregateOptions) -> Self {
+        let mut document = bson::Document::new();
+
+        if let Some(allow_disk_use) = options.allow_disk_use {
+            document.insert("allowDiskUse", Bson::Boolean(allow_disk_use));
+        }
+
+        // useCursor not currently used by the driver.
+
+
+        let cursor = doc! { "batchSize" => (options.batch_size) };
+        document.insert("cursor", Bson::Document(cursor));
+
+        // maxTimeMS is not currently used by the driver.
+
+        // read_preference is used directly by Collection::aggregate.
+
+        document
+    }
+}
+
 /// Options for count queries.
-#[derive(Clone, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct CountOptions {
-    pub skip: u64,
-    pub limit: i64,
+    pub skip: Option<i64>,
```

Comment:
I'm not quite sure why an unsigned integer was used here (and in another place below). The spec doesn't use unsigned integers (probably due to there not being an easy why to enforce it in dynamically typed languages)